src/s3_sup.erl

```diff
@@ -0,0 +1,46 @@
+%%% @doc
+%%% Top level supervisor for the S3 client app.
+%%% @end
+-module(s3_sup).
+
+-behaviour(supervisor).
+
+%% API
+-export([start_link/0]).
+
+%% Supervisor callbacks
+-export([init/1]).
+
+-define(SERVER, ?MODULE).
+
+
+%%% API functions
+
+%% @doc
+%% Starts the supervisor.
+%% @end
+start_link() ->
+    supervisor:start_link({local, ?SERVER}, ?MODULE, []).
+
+%%% Supervisor callbacks
+
+%% @private
+%% @doc
+%% Start an instance of s3_server.
+%% @end
+init([]) ->
+    SupFlags = {one_for_one, 1000, 3600},
```

Comment:
I guess we are keeping compatiblity with OTP < 17? Otherwise `maps` format is preferred and tuple is deprecated.
Question no 2 - how did you chose these restart policy values?

Reply:
Yes, if anyone is still running older ERTS but wants to leverage this change they can.

Reply:
They were arbitrarily chosen, I don't think there is any cases where the workers would crash s3_server to restart it. though I'm up for a better suggestion than these values but we have been running with it for a while for some of our systems.